### DISCUSSION TITLE
How can i use the Kratos ?

### DISCUSSION THREAD

**Discussion by jhhan9090** [KratosMultiphysics/Kratos#11676] [General]
Hi,
I'm new to using the Kratos

Especially, i'm interested in DEM simulation between the contact surfaces when the frictional interaction occurs.
During installing the Kratos, I had some problems.

firstly, I try to compile with code uploaded on this site.
Compiling failed even if I installed additional apps(Boost, Cmake, etc.)
Then, I installed the Kratos based on pip and GiD.
This method meets the license problem for GiD.

Maybe there is a method to operate the entire code properly without any license problem
is there anyone who can give me advice on using Kratos?

1) is there any difference between compiling and pip-based installation when i do simulation?
2) Maybe there is a license problem for GiD, I don't want to use it. then is there any other method for pre-post processing?

Thank you !

**Comment by AlejandroCornejo** [comment]
Good morning,

Why compilations failed? it should work following the https://github.com/KratosMultiphysics/Kratos/blob/master/INSTALL.md

Are you working in linux or windows?

Answering your questions:
1. No, the only difference is that compiling allows you to use the very latest version of the code. The pip is updated from time to time.
2. Some people use SALOME preprocessor for this purpose. However, the DEM team works mainly in GiD @salvalatorre @ChengshunShang1996

**Reply by AlejandroCornejo** [reply]
@KratosMultiphysics/dem

**Reply by jhhan9090** [reply]
Thank you for your kind answers :)

Today, I finished the entire installation.
I'm testing how it works for DEM with some examples
These codes are very interesting and well constructed

While working with some examples, the question about the license came out again.
For the GiD, the academic license has been assigned to my PC now.
I installed the developer version of GiD

Is there any limitation to model large size(numerous particles in DEM) for academic license?
I need to model a huge size model, maybe later i will simulate model in a mesoscale.

again, Thank you for your replying :)

**Reply by ChengshunShang1996** [reply]
Hi Han. I am glad to know that you are interested in Kratos DEM. Although GiD is usually used for pre and post processing by our team, the main code of the Kratos DEMApplication is independent of GiD. So you can build a model with any model size and particle number.

**Reply by AlejandroCornejo** [reply]
AFAIK you can renew the GID student license up to 2 times more (3 months in total) with no limitation in terms of elements. With no license the limit is 10 k elements

**Reply by vaishnavi9247263609** [reply]
Hi @AlejandroCornejo, I'm trying to run it on Windows. I've followed the same steps mentioned in https://github.com/KratosMultiphysics/Kratos/blob/master/INSTALL.md. But, I'm getting "142 warnings generated.
ninja: build stopped: subcommand failed.".

**Reply by AlejandroCornejo** [reply]
If you build in Windows you should use the Visual Studio compiler, why do you have this ninja warning? @roigcarlo

**Reply by roigcarlo** [reply]
As @AlejandroCornejo says, you should not use ninja in windows. Also please ignore the mingw instructions unless you have heavy restrictions and cannot use the VS compiler.

The sections you must follow are:

https://github.com/KratosMultiphysics/Kratos/blob/master/INSTALL.md#windows-installation
https://github.com/KratosMultiphysics/Kratos/blob/master/INSTALL.md#windows
https://github.com/KratosMultiphysics/Kratos/blob/master/INSTALL.md#windows-1

**Reply by PouyaCR** [reply]
Hello
I used GiD problem type to download keratos. But, I cannot find PFEM section.
![image](https://github.com/KratosMultiphysics/Kratos/assets/145875446/ebb7cb71-2ec9-4332-82bd-47d82e81ea2b)

**Reply by rubenzorrilla** [reply]
@jginternational

**Reply by jginternational** [reply]
Hi @PouyaCR 

I can see that you are using the GUI available in GiD. In those releases we do not pack all the applications.

If you want to use all the implemented applications, you may be interested in the developer version.
Follow the first 3 steps in the repository 
https://github.com/KratosMultiphysics/GiDInterface

Javi Gárate
